druid-shell/src/platform/windows/window.rs

```diff
@@ -1003,25 +1060,13 @@ impl WndProc for MyWndProc {
                             s.handler.mouse_down(&event);
                         } else {
                             s.handler.mouse_up(&event);
-                            should_release_capture = s.exit_mouse_capture(button);
+                            if s.exit_mouse_capture(button) {
+                                self.handle.borrow().defer(DeferredOp::ReleaseMouseCapture);
+                            }
```

Comment:
I'm curious what happens if we continue to release here like normal? Would we have already handled the deferred `ContextMenu` action?

Reply:
Prior to this PR, we would do the context menu synchronously in `handler.mouse_up` above. As it is in this PR right now, we run the context menu just below this, at the closing brace on line 1067.

The latest changes to `ReleaseCapture` don't actually change the ordering between the context menu and releasing the capture. Before the changes, we ran the context menu when dropping the handle borrow, and then a few lines later we have the release capture. After the changes, we run the context menu *and* the release capture (in that order) when dropping the handle borrow. The main reason for the changes to `ReleaseCapture` in the latest patch is that we have a general mechanism for deferring now, so I figured we should use it. The only actual behavior change is the silencing of the warning.

Reply:
i'll experiment a bit.